dali/operators/python_function/dltensor_function.h

```diff
@@ -258,7 +258,10 @@ class DLTensorPythonFunctionImpl : public Operator<Backend> {
   }
 
   ~DLTensorPythonFunctionImpl() {
-    auto interpreter_lock = py::gil_scoped_acquire();
+    try {
+      auto interpreter_lock = py::gil_scoped_acquire();
+    } catch (...) {
+    }
```

Comment:
This code is broken now. This "lock" is like a lock guard. It goes out of scope and the lock is released; then we do everything outside of the lock.

Also, I don't think that letting this error just disappear is a good idea.

Reply:
Wouldn't this clear up anyway since this is in the `~DLTensorPythonFunctionImpl` ?
I considered suggesting that this should kill the whole thing but I am not sure that it can occur?

Reply:
Yeah, this should only throw in some extreme cases. If we cannot access GIL there's probably something more wrong with the process so maybe it's just better to let it terminate